contrib/rpmbuild/src/python/pants/contrib/rpmbuild/tasks/rpmbuild_task.py

```diff
@@ -0,0 +1,362 @@
+                  self.write_stream(fileobj, f)
+                output_dir = os.path.join(self.get_options().pants_distdir, 'rpmbuild', rpmdir)
+                safe_mkdir(output_dir)
+                shutil.copy(rpmfile, output_dir)
+                if name.startswith('home/rpmuser/rpmbuild/RPMS/'):
+                  self.context.products.get('rpms').add(vt.target, vt.results_dir).append(rel_rpm_path)
+                else:
+                  self.context.products.get('srpms').add(vt.target, vt.results_dir).append(rel_rpm_path)
+
+        retcode = proc.wait()
+        if retcode != 0:
+          raise TaskError('Failed to extract RPMS')
+        else:
+          # Save the resulting image if asked. Eventually this image should be pushed to the registry every build,
+          # and subsequent invocations on the published RPM should simply pull and extract.
+          if self.get_options().commit_container_image:
+            commited_name = 'rpm-commited-image-{}'.format(uuid_identifier)
+            self.context.log.info('Saving container state as image...')
+            docker_commit_cmd = [self.get_options().docker, 'commit', container_name]
+            with self.docker_workunit(name='commit-to-image', cmd=docker_commit_cmd) as workunit:
+              subprocess.call(docker_commit_cmd, stdout=workunit.output('stdout'), stderr=subprocess.STDOUT)
+              self.context.log.info('Saved container as image: {}\n'.format(commited_name))
+
+    finally:
+      # Remove the build container.
+      if container_name and not self.get_options().keep_build_products:
+        remove_container_cmd = [self.get_options().docker, 'rm', container_name]
+        with self.docker_workunit(name='remove-build-container', cmd=remove_container_cmd) as workunit:
+          subprocess.call(remove_container_cmd, stdout=workunit.output('stdout'), stderr=subprocess.STDOUT)
+
+      # Remove the build image.
+      if not self.get_options().keep_build_products:
+        remove_image_cmd = [self.get_options().docker, 'rmi', image_name]
+        with self.docker_workunit(name='remove-build-image', cmd=remove_image_cmd) as workunit:
+          subprocess.call(remove_image_cmd, stdout=workunit.output('stdout'), stderr=subprocess.STDOUT)
+
+  def execute(self):
+    platform_key = self.get_options().platform
+    try:
+      platform = self.get_options().platform_metadata[platform_key]
+      platform['id'] = platform_key
+    except KeyError:
+      raise TaskError('Unknown platform {}'.format(platform_key))
+
+    targets = self.context.targets(self.is_rpm_package)
+    with self.invalidated(targets, invalidate_dependents=True, topological_order=True) as invalidation_check:
+      # Use of invalidation to establish workdir and to get dependency management - all targets in context are built.
+      for vt in invalidation_check.all_vts:
+        # TODO(mateo): Hooking up to the DockerPlatform will make iterating a lot less painful, since we can publish
```

Comment:
I guess kill this comment since that work is up in the air ATM.

Reply:
Will do.